Add rdbms-postgresql driver which uses hu.dwim.rdbms

darabi opened the pull request:
The library hu.dwim.rdbms supports several database backends
(PostgreSQL, SQLite, and Oracle) through a uniform interface. The
PostgreSQL backend uses Postmodern's [1] cl-postgres [2] library.
    
Tests from test-suite.lisp are split in separate files.
    
[1] https://marijnhaverbeke.nl/postmodern/
[2] https://marijnhaverbeke.nl/postmodern/cl-postgres.html

dnaeon commented on src/driver/rdbms-postgresql.lisp:
```diff
@@ -0,0 +1,172 @@
+                 :provider provider
+                 :connection-specification connection-specification))
+
+;; (defparameter *tstdrv* (make-driver (cl-migratum.provider.local-path:make-local-path-provider "~/common-lisp/cl-migratum/t/migrations/") '(:database "migratum" :user-name "migratum" :password "FvbRd5qdeWHNum9p")))
```
Perhaps a leftover from some testing?

dnaeon replied:
Can you please remove the leftovers from PR, as they are still here?

dnaeon commented:
Also, could you please update the README and specify that this driver is also available?

darabi commented:
Will do that after I've seen a green GitHub Actions build on my branch. Will inform you when done.

darabi commented:
I added the README changes and fixed the build. GitHub currently has problems with Actions (https://www.githubstatus.com/) but there was a green test run before I force-pushed the README file, so I'm quite confident that everything is OK.

dnaeon commented on cl-migratum.driver.rdbms-postgresql.asd:
```diff
@@ -0,0 +1,23 @@
+(defsystem "cl-migratum.driver.rdbms-postgresql"
+  :name "cl-migratum.driver.rdbms-postgresql"
+  :description "cl-migratum driver for driving migrations against SQL databases using hu.dwim.rdbms.postgresql"
+  :version "0.3.0"
```
Generally I go with separate version of the drivers and providers, as they might evolve on their own. That way the core framework (`cli-migratum`) might have one version, but a driver could be on a different version.

Since this is a new driver, I think it makes sense to have it as `0.1.0` version.

dnaeon commented on cl-migratum.driver.rdbms-postgresql.asd:
```diff
@@ -0,0 +1,23 @@
+  :name "cl-migratum.driver.rdbms-postgresql"
+  :description "cl-migratum driver for driving migrations against SQL databases using hu.dwim.rdbms.postgresql"
+  :version "0.3.0"
+  :author "Marin Atanasov Nikolov <dnaeon@gmail.com>"
```
I think it's fair to have you here instead of me, since you've created this new driver :)

dnaeon commented on src/driver/rdbms-postgresql.lisp:
```diff
@@ -0,0 +1,172 @@
+;; Copyright (c) 2020-2022 Marin Atanasov Nikolov <dnaeon@gmail.com>
```
Please consider adding your name here as well, since you've authored it.

darabi commented:
Version is now at 0.1.0, added my name in `defsystem` and to the copyright notice.

dnaeon commented:
@darabi , thanks!
